### ISSUE TITLE
Test UI should surface continuous run on gutter test icon

### ISSUE THREAD

**Issue by JustinGrote** [microsoft/vscode#188305]
CC @connor4312 

The `Turn On Continuous Run` option should also be surfaced in the right click menu for the gutter icons

![image](https://github.com/microsoft/vscode/assets/15258962/73e7ee8c-d0de-4186-b55f-5c6979cb9edf)

![image](https://github.com/microsoft/vscode/assets/15258962/1c8220c1-5a4f-43be-9903-5209509af4fa)

**Comment by weartist** [comment]
I'll try to fix this

**Comment by weartist** [comment]
@connor4312 Hi, is this issue not accepting PR?

**Comment by connor4312** [comment]
We would, I just don't think the solution of what to display is very obvious so I removed the tag

**Comment by weartist** [comment]
> We would, I just don't think the solution of what to display is very obvious so I removed the tag. Ideas/contributions welcome though.

Thanks!, By the way, isn't it only issues with the help want tag that accept external PRs? That's what I've always thought XD

**Comment by connor4312** [comment]
Not necessarily. Personally I use the tag when the issue has a clear+scoped solution and is not something I'm going to get to in the immediate future.

**Comment by JustinGrote** [comment]
Why wouldn't it just be exposed as `Turn on Continuous Run` at the gutter level and then hook to the same function/event? What am I missing that's non-obvious?

**Comment by connor4312** [comment]
I think that if we have the context menu option, we should also reflect the state of continuous run being on on that icon as well, which is the trickier part. I would take a PR adding the action alone, but would want the state to be reflected before I would consider this "complete"

**Comment by JustinGrote** [comment]
The status would be reflected in the test explorer, and conversely when you flag a test for continuous run in the test explorer it doesn't update the gutter, either way though it probably would be good to implement if things are changing anyways.

My thoughts:
1. Make the gutter icon the same `eye` codicon upon activation, with color still matching the last test result (light green/green/red)
2. All other "processing" occurs as normal

Issue:
1. What if a test is in suspended state or otherwise where the color conflicts with an existing status color? How do we represent a test is both `watched` and `suspended` or do we care?


Alternatives:
1. Some alternate indicator that doesn't affect the gutter icon directly but would be next to it. Highlighted line maybe?

**Comment by sheremet-va** [comment]
The visual "eye" syncing has been working since https://github.com/microsoft/vscode/issues/210848#issuecomment-2499273293 via `vscode.startContinuousTestRun` command. We can even get the needed `testItem` when clicking on the custom command in the test gutter, but we don't have access to test item information in the when context, so it's not possible to display different commands like "turn on"/"turn off".

@connor4312 do you think this toggle can be implemented in vscode itself, or can we get some information about the test item in the when context? (maybe at least id?)

**Comment by connor4312** [comment]
Yea, this should be implemented in VS Code itself. PR would be welcome if anyone is feeling ambituous. Code pointers:

https://github.com/microsoft/vscode/blob/f12fa05bd6e9b42ace6824280049b395cbbd0723/src/vs/workbench/contrib/testing/common/testingContinuousRunService.ts#L25

Would update the display here:

https://github.com/microsoft/vscode/blob/f12fa05bd6e9b42ace6824280049b395cbbd0723/src/vs/workbench/contrib/testing/browser/testingDecorations.ts#L917
